src/main/scala/chisel3/util/experimental/ForceNames.scala

```diff
@@ -0,0 +1,290 @@
+// See LICENSE for license details.
+
+package chisel3.util.experimental
+
+import chisel3.experimental.{ChiselAnnotation, RunFirrtlTransform, annotate}
+import firrtl.Mappers._
+import firrtl._
+import firrtl.annotations._
+import firrtl.annotations.TargetToken._
+import firrtl.ir._
+import firrtl.analyses.InstanceKeyGraph
+import firrtl.options.Dependency
+import firrtl.passes.{InlineInstances, LowerTypes}
+import firrtl.stage.Forms
+import firrtl.stage.TransformManager.TransformDependency
+
+import scala.collection.mutable
+
+object forceName {
+
+  /** Force the name of this signal
+    * @param signal Signal to name
+    * @param name Name to force to
+    */
+  def apply(signal: => chisel3.Bits, name: String): Unit = {
```

Comment:
`Bits` or can we do `Element`?
```suggestion
  def apply[T <: chisel3.Element](signal: T, name: String): T = {
```
A few things in this suggestion:
* `Element` over `Bits` - Allow for marking `Clock`, `Reset`, and `Analog`.
* eager vs. by-name parameter - Because this doesn't evaluate the signal, bad things could happen if you had hardware in this. That being said, since we don't currently return it, it's less likely.
* Returning the `signal` - helpful in not having to have an extra Expression

Not saying all of these are needed, just think all 3 are worth considering.

Reply:
Sam suggestions for all APIs below